src/Type/Php/DateTimeSubMethodThrowTypeExtension.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use DateTime;
+use DateTimeImmutable;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Php\PhpVersion;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\DynamicMethodThrowTypeExtension;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use function count;
+use function in_array;
+
+final class DateTimeSubMethodThrowTypeExtension implements DynamicMethodThrowTypeExtension
+{
+
+	public function __construct(private PhpVersion $phpVersion)
+	{
+	}
+
+	public function isMethodSupported(MethodReflection $methodReflection): bool
+	{
+		return $methodReflection->getName() === 'sub'
+			&& in_array($methodReflection->getDeclaringClass()->getName(), [DateTime::class, DateTimeImmutable::class], true);
+	}
+
+	public function getThrowTypeFromMethodCall(MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope): ?Type
+	{
+		if (count($methodCall->getArgs()) === 0) {
+			return null;
+		}
+
+		if (!$this->phpVersion->hasDateTimeExceptions()) {
+			return null;
+		}
+
+		return new ObjectType('DateMalformedStringException');
```

Comment:
`DateTime::sub` does not accepts constant strings, the signature is
```
public DateTime::sub(DateInterval $interval): DateTime
```

It only throw an exception if the DateInterval is relative like
```
$dt->sub(new DateInterval('next weekday'))
```
but I didnt see a way to have something like `getDateIntervalConstantStrings()`. I wasn't sure
- if it would require to introduce a DateIntervalType like the QueryType (https://github.com/phpstan/phpstan-doctrine/blob/38db3bad8f1567d7bf64806738d724261f8a2b5c/src/Type/Doctrine/Query/QueryType.php#L11) 
- if I was able to reproduce the implementation...

So as a first step, it improve the situation for all the PHP version lower than 8.3.